rsql-jpa/src/test/java/io/github/perplexhub/rsql/RSQLJPASupportTest.java

```diff
@@ -527,6 +527,45 @@ final void testNotLike() {
 		assertThat(rsql, count, is(6L));
 	}
 
+	@Test
+	final void testEscapeForUnderscore() {
+		char escapeChar = '$';
+		QuerySupport query = QuerySupport.builder()
+				.rsqlQuery("code=like='" + escapeChar + "_'")
+				.escapeCharacter(escapeChar)
+				.build();
+		List<Company> users = companyRepository.findAll(toSpecification(query));
+		long count = users.size();
+		log.info("rsql: {} -> count: {}", query.getRsqlQuery(), count);
+		assertThat(query.getRsqlQuery(), count, is(2L));
+	}
+
+	@Test
+	final void testEscapeForPercent() {
+		char escapeChar = '$';
+		QuerySupport query = QuerySupport.builder()
+				.rsqlQuery("name=like='" + escapeChar + "%'")
+				.escapeCharacter(escapeChar)
+				.build();
+		List<Company> users = companyRepository.findAll(toSpecification(query));
+		long count = users.size();
+		log.info("rsql: {} -> count: {}", query.getRsqlQuery(), count);
+		assertThat(query.getRsqlQuery(), count, is(2L));
+	}
+
+	@Test
+	final void testEscapeForAntiSlash() {
+		char escapeChar = '\\';
+		QuerySupport query = QuerySupport.builder()
+				.rsqlQuery("name=like='\\\\%'")
+				.escapeCharacter(escapeChar)
+				.build();
+		List<Company> users = companyRepository.findAll(toSpecification(query));
+		long count = users.size();
+		log.info("rsql: {} -> count: {}", query.getRsqlQuery(), count);
+		assertThat(query.getRsqlQuery(), count, is(2L));
+	}
+
```

Comment:
Couple of suggestions here:

- Can we make tests to save their own data and make the test method transactional in other to not affect other tests. Because now if we insert some test data we have to go over all tests and fix them.
- If we asserting only count we should query count only. E.g. `companyRepository.count(toSpecification(query))`

Reply:
Sure, I can't figure why I didn't do that way.